walker.go

```diff
@@ -0,0 +1,84 @@
+package ftp
+
+import (
+	"fmt"
+	"strings"
+)
+
+//Walker traverses the directory tree of a remote FTP server
+type Walker struct {
+	serverConn *ServerConn
+	root       string
+	cur        item
+	stack      []item
+	descend    bool
+}
+
+type item struct {
+	path  string
+	entry Entry
+	err   error
+}
+
+// Step advances the Walker to the next file or directory,
+// which will then be available through the Path, Stat, and Err methods.
+// It returns false when the walk stops at the end of the tree.
+func (w *Walker) Step() bool {
```

Comment:
Better call the function `Next()` like many other iterable objects in go.

Reply:
Made the change before I saw your edit :-). I think that whilst `Next()` is the right name to use for iterable objects, `Step()` does seem to be the name for directory walking so for consistency I think `Step()` should be used